### PR TITLE
Created models and basic functions for tests

### PR DESCRIPTION
**Pull request by KozielPiotr**

### CODE REVIEW

File: bills_app/apps.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
+"""bills_app config."""
 from django.apps import AppConfig
 
 
 class BillsAppConfig(AppConfig):
+    """bills_app config."""
     name = 'bills_app'
```
Comment (cmiacz) [inline bills_app/apps.py]: Keep names simple. Sufix `_app` is obsolete in case of application. Rename to `bills`

File: bills_app/models.py [hunk 2]
```diff
@@ -1,3 +1,47 @@
+"""Models for bills_app."""
+
+from django.contrib.auth.models import AbstractUser
 from django.db import models
 
-# Create your models here.
+
+class CustomUser(AbstractUser):
```
Comment (cmiacz) [inline bills_app/models.py]: Please create separate `Participant` model for billing. It's not needed to mix bill participant with django authentication user model. Do not override default model if not necessary.

File: bills_app/models.py [hunk 3]
```diff
@@ -1,3 +1,47 @@
...
+    name = models.CharField(max_length=500, unique=True)
+    cashier = models.ForeignKey(CustomUser, null=True, blank=True, on_delete=models.SET_NULL,
+                                related_name="cashier")
+    participators = models.ManyToManyField(CustomUser, related_name="participates")
+
+    def __str__(self):
+        return self.name
+
+
+class Receipt(models.Model):
```
Comment (cmiacz) [inline bills_app/models.py]: Keep naming consistent. Rename to `Bill` as the whole project is called `shared_bills` and receipt this is just the bill to share.

File: bills_app/models.py [hunk 4]
```diff
@@ -1,3 +1,47 @@
...
+    """Single bill, eg. for car, food, fuel etc."""
+    for_what = models.CharField(max_length=100)
```
Comment (cmiacz) [inline bills_app/models.py]: Rename to `title`. Please also consider to add additional `description` field

File: bills_app/models.py [hunk 5]
```diff
@@ -1,3 +1,47 @@
...
+    amount = models.DecimalField(max_digits=7, decimal_places=2)
```
Comment (cmiacz) [inline bills_app/models.py]: Consider using `MoneyField` from `django-money`
https://github.com/django-money/django-money

File: bills_app/models.py [hunk 6]
```diff
@@ -1,3 +1,47 @@
...
+    event = models.ForeignKey(Event, on_delete=models.PROTECT)
```
Comment (cmiacz) [inline bills_app/models.py]: Bills should be automatically deleted when the whole event is being removed. Use `on_delete=models.CASCADE`

File: bills_app/models.py [hunk 7]
```diff
@@ -1,3 +1,47 @@
...
+    def __str__(self):
+        return self.username
+
+
+class Event(models.Model):
+    """Settlement for whole event, eg. kayaks."""
+    name = models.CharField(max_length=500, unique=True)
+    cashier = models.ForeignKey(CustomUser, null=True, blank=True, on_delete=models.SET_NULL,
+                                related_name="cashier")
+    participators = models.ManyToManyField(CustomUser, related_name="participates")
```
Comment (cmiacz) [inline bills_app/models.py]: Remove this relation. Model `Participant` should have foreign key to `Event` with backref called `participants` as participants exists only in scope of event and should not be shared between different events.

File: bills_app/models.py [hunk 8]
```diff
@@ -1,3 +1,47 @@
...
+    """
+
+    def __str__(self):
+        return self.username
+
+
+class Event(models.Model):
+    """Settlement for whole event, eg. kayaks."""
+    name = models.CharField(max_length=500, unique=True)
+    cashier = models.ForeignKey(CustomUser, null=True, blank=True, on_delete=models.SET_NULL,
```
Comment (cmiacz) [inline bills_app/models.py]: Rename to `paymaster`

File: bills_app/models.py [hunk 9]
```diff
@@ -1,3 +1,47 @@
...
+    def __str__(self):
+        return self.name
+
+
+class Receipt(models.Model):
+    """Single bill, eg. for car, food, fuel etc."""
+    for_what = models.CharField(max_length=100)
+    amount = models.DecimalField(max_digits=7, decimal_places=2)
+    event = models.ForeignKey(Event, on_delete=models.PROTECT)
+    paid_by = models.ForeignKey(CustomUser, null=True, blank=True, on_delete=models.PROTECT,
```
Comment (cmiacz) [inline bills_app/models.py]: Rename to `payer`

File: bills_app/models.py [hunk 10]
```diff
@@ -1,3 +1,47 @@
...
+
+    def __str__(self):
+        return self.for_what
+
+
+class Payment(models.Model):
+    """Payment from one to another."""
+    who_pays = models.ForeignKey(CustomUser, on_delete=models.CASCADE, related_name="paid_cash")
+    for_who = models.ForeignKey(CustomUser, on_delete=models.CASCADE, related_name="got_cash")
+    event = models.ForeignKey(Event, on_delete=models.PROTECT)
```
Comment (cmiacz) [inline bills_app/models.py]: `on_delete=models.CASCADE`

File: bills_app/models.py [hunk 11]
```diff
@@ -1,3 +1,47 @@
...
+                                related_name="paid_for")
+    bought_by = models.ManyToManyField(CustomUser, related_name="bought")
+
+    def __str__(self):
+        return self.for_what
+
+
+class Payment(models.Model):
+    """Payment from one to another."""
+    who_pays = models.ForeignKey(CustomUser, on_delete=models.CASCADE, related_name="paid_cash")
```
Comment (cmiacz) [inline bills_app/models.py]: Rename to `issuer`

File: bills_app/models.py [hunk 12]
```diff
@@ -1,3 +1,47 @@
...
+    for_who = models.ForeignKey(CustomUser, on_delete=models.CASCADE, related_name="got_cash")
```
Comment (cmiacz) [inline bills_app/models.py]: Rename to `acquirer`

File: bills_app/models.py [hunk 13]
```diff
@@ -1,3 +1,47 @@
...
+
+
+class Receipt(models.Model):
+    """Single bill, eg. for car, food, fuel etc."""
+    for_what = models.CharField(max_length=100)
+    amount = models.DecimalField(max_digits=7, decimal_places=2)
+    event = models.ForeignKey(Event, on_delete=models.PROTECT)
+    paid_by = models.ForeignKey(CustomUser, null=True, blank=True, on_delete=models.PROTECT,
+                                related_name="paid_for")
+    bought_by = models.ManyToManyField(CustomUser, related_name="bought")
```
Comment (cmiacz) [inline bills_app/models.py]: Rename to `participants`

**Review by cmiacz — CHANGES_REQUESTED** [review]

**Comment by cmiacz** [comment]
Please add migrations to repo

File: bills_app/models.py (hunk 7 above)
Comment (KozielPiotr) [reply bills_app/models.py]: I'd like to keep this one unchanged. After once created, the participant could be assigned to more than one event.

Comment (cmiacz) [reply bills_app/models.py]: In that case participant should be somehow assigned to user. We don't want to share participant names across all application. As we don't have any authentication schema now, plaese keep participants assigned to events with foreign key.

ManyToMany relation will be useful in the future as we plan to implement sharing feature. Each event will then have a foreign key to his owner (django authenticated user) and many to many realtion to other users allowed to access event with permissions specified by the owner

File: bills/models.py [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+    """Payment from one to another."""
+
+    title = models.CharField(max_length=500)
+    issuer = models.ForeignKey(
+        Participant, on_delete=models.CASCADE, related_name="paid"
+    )
+    acquirer = models.ForeignKey(
+        Participant, on_delete=models.CASCADE, related_name="acquired"
+    )
+    event = models.ForeignKey(Event, on_delete=models.PROTECT)
```
Comment (cmiacz) [inline bills/models.py]: CASCADE

File: bills/models.py [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+    def __str__(self):
+        return self.title
+
+
+class Payment(models.Model):
+    """Payment from one to another."""
+
+    title = models.CharField(max_length=500)
+    issuer = models.ForeignKey(
+        Participant, on_delete=models.CASCADE, related_name="paid"
```
Comment (cmiacz) [inline bills/models.py]: PROTECT. Participant cannot be deleted if there are any related payments

File: bills/models.py [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+    )
+    acquirer = models.ForeignKey(
+        Participant, on_delete=models.CASCADE, related_name="acquired"
```
Comment (cmiacz) [inline bills/models.py]: PROTECT. Participant cannot be deleted if there are any related payments

File: bills/tests.py [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+    :param amount: decimal with price
+    :param event: Event object to make relationship
+    :param payer: Participant object to make relationship
+    :param bought_by: list of Participant objects to make relationship
+    :return: new Bill object
+    """
+    bill = Bill.objects.create(
+        title=title, balance=Money(amount, "PLN"), event=event, payer=payer
+    )
+    bill.bought_by.add(bought_by)
```
Comment (cmiacz) [inline bills/tests.py]: It seems to be outdated

**Review by cmiacz — CHANGES_REQUESTED** [review]

File: shared_bills/settings.py [hunk 18]
```diff
@@ -20,7 +20,7 @@
 # See https://docs.djangoproject.com/en/3.0/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = 'e45i2uj&!e(8#-!_#n6z^vmj+o^dze8g^!jb!n2c4zxif&0r&@'
+SECRET_KEY = "e45i2uj&!e(8#-!_#n6z^vmj+o^dze8g^!jb!n2c4zxif&0r&@"
```
Comment (cmiacz) [inline shared_bills/settings.py:23]: DO NOT COMMIT SECRET KEY TO REPO
It should be obtained from environment variable

**Review by cmiacz — CHANGES_REQUESTED** [review]

**Review by cmiacz — APPROVED** [review]

**Pull request merged**